clang/include/clang/CIR/FrontendAction/CIRGenAction.h

```diff
@@ -55,6 +56,13 @@ class EmitCIRAction : public CIRGenAction {
   EmitCIRAction(mlir::MLIRContext *MLIRCtx = nullptr);
 };
 
+class EmitLLVMAction : public CIRGenAction {
+  virtual void anchor();
+
+public:
+  EmitLLVMAction(mlir::MLIRContext *MLIRCtx = nullptr);
```

Comment:
Under what circumstances does it make sense to pass a null MLIR context? (Mostly wondering why the pattern is to use a default parameter here rather than make the caller pass it explicitly.)

Reply:
This seems to be following a pattern established by CodeGenAction, and calling this without a context parameter is the normal case (in fact, it's the only case I see in the code). When the parameter is null, the CIRGenAction constructor allocates a new context. If a non-null pointer is passed, the constructor just uses it.